plugins/CoreAdminHome/Commands/InvalidateReportData.php

```diff
@@ -223,4 +239,40 @@ private function getSegmentsToInvalidateFor(InputInterface $input, $idSites)
         }
         return $result;
     }
+
+    private function getSegmentsIdToInvalidateFor(InputInterface $input, $idSites)
+    {
+        $idSegments = $input->getOption('idsegments');
+        $idSegments = trim($idSegments);
+        $idSegments = preg_replace('/\s/', '', $idSegments);
+        $idSegments = explode(",", $idSegments);
+        $idSegments = array_map('trim', $idSegments);
+        $idSegments = array_unique($idSegments);
+
+        if (empty($idSegments)) {
+            return array(null);
+        }
+
+        /** @var SegmentEditorModel $segmentEditorModel */
+        $segmentEditorModel = StaticContainer::get('Piwik\Plugins\SegmentEditor\Model');
+        $segments = $segmentEditorModel->getAllSegmentsAndIgnoreVisibility();
+
+        $segments = array_filter($segments, function ($segment) use ($idSegments) {
+            return in_array($segment['idsegment'], $idSegments);
+        });
+
+        $segments = array_map(function ($segment) {
+            return $segment['definition'];
+        }, $segments);
+
+        $result = array();
+        foreach ($segments as $segmentString) {
+            try {
+                $result[] = new Segment($segmentString, $idSites);
+            } catch (\Exception $e) {
+                return $e->getMessage();
```

Comment:
we'd actually expect the exception not to be caught just like in the current `getSegmentsToInvalidateFor()` method. This way user will know there is an invalid segment. 
Otherwise you would for example return a `string` here instead of an `array` and the code above would not work correctly anymore.